datafusion/core/src/physical_plan/planner.rs

```diff
@@ -2053,10 +2053,7 @@ mod tests {
         ];
         for case in cases {
             let logical_plan = test_csv_scan().await?.project(vec![case.clone()]);
-            let message = format!(
-                "Expression {case:?} expected to error due to impossible coercion"
-            );
-            assert!(logical_plan.is_err(), "{}", message);
```

Comment:
Type checking is done in the `Type coercion analyzer rule` rather than directly in the build